Sources/StreamChat/Models/Channel+Sorting.swift

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// last_updated, last_message_at, updated_at, created_at, member_count, unread_count or has_unread
+
+class Example {
+    let channels: [ChatChannel] = []
+
+    func sortWithDynamicMember(with sorting: [Sorting<ChannelListSortingKey>]) -> [ChatChannel] {
```

Comment:
With dynamicMemberLookup, we should be able to use the existing combination of `Sorting` plus `ChannelListSortingKey `. The only problem is that ChannelListSortingKey is an enum, and does not allow custom fields. We could probably just create a new `CustomSortingKey` struct, that can be used for it 🤔 .

Reply:
This solution comes at the expense of the implementation of `subscript[dynamicMember: ]`, though

Reply:
I don't think that it is a big deal if we break a little bit the API here but make it "compatible". What I mean is that we can convert the enum to a struct with static properties like we do in other places.

This would allow for customers to do something like this:
`.custom("test", \ChatChannel.test)`

This will also work for extra data since customers can extend `ChatChannel` like so:
```swift
extension ChatChannel {
    var test: String {
        extraData["test"]?.stringValue ?? ""
    }
}
```